routing/dht/handlers.go

```diff
@@ -226,5 +226,5 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 		dht.providers.AddProvider(ctx, key, p)
 	}
 
-	return pmes, nil // send back same msg as confirmation.
```

Comment:
hmm so then we really have no way of confirming. could we maybe comment this out and make note of this? from a protocol standpoint it will be useful to confirm these things-- but perhaps the protocol can change.

By the way, muxing all dht traffic over one stream will really, really help.